File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        long visited = 0;
        long multipleVisited = 0;
        int n = s.length();
        int p;
        int x;
        for(int i=0;i<n;i++) {
            p = s[i] - 'a';
            x = (1 << p);
            if((visited & x) == 0) {
                visited =  visited | x;
            } else {
                multipleVisited = multipleVisited | x;
            }
        }
        for(int i=0;i<n;i++) {
            int p = s[i] - 'a';
            if(((visited & (1 << p)) != 0) && ((multipleVisited & (1 << p )) == 0))
                return i;
        }
        return -1;
    }

};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int a[] = new int[]{-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1};
        for(int i=0;i<s.length();i++) {
            int p = s.charAt(i)-'a';
            if(a[p] == -1)
                a[p] = i;
            else
                a[p] = -2;
        }
        int min = -1;
        for(int i=0;i<26;i++) {
            if(a[i]>=0) {
                if(min == -1 || a[min] > a[i])
                    min = i;
            }
        }
        return min == -1?-1:a[min];
    }
}